src/client/clientlauncher.cpp

```diff
@@ -53,17 +53,13 @@ MainGameCallback *g_gamecallback = NULL;
 
 ClientLauncher::~ClientLauncher()
 {
-	if (receiver)
-		delete receiver;
+	delete receiver;
 
-	if (input)
-		delete input;
+	delete input;
 
-	if (g_fontengine)
-		delete g_fontengine;
+	delete g_fontengine;
 
-	if (device)
```

Comment:
Okay, now I see the point. Got confused with uni stuff in plain C. We definitely do need to check before calling. How about this?

```cpp
if (device)
 	if(!device->drop())
 		// error
```